tests/table/tables_plugin.test.ts

```diff
@@ -0,0 +1,836 @@
+      setCellContent(model, "C3", "Something Else");
+      expect(zoneToXc(model.getters.getTables(sheetId)[0].zone)).toEqual("B1:C3");
+      expect(getFilterValues(model)).toEqual([
+        { zone: "B1:B3", value: ["C"] },
+        { zone: "C1:C3", value: [] },
+      ]);
+    });
+
+    test("Table zone isn't expended at creation", () => {
+      setCellContent(model, "A4", "Something");
+      createTable(model, "A1:A3");
+      expect(zoneToXc(model.getters.getTables(sheetId)[0].zone)).toEqual("A1:A3");
+      expect(getFilterValues(model)).toEqual([{ zone: "A1:A3", value: [] }]);
+    });
+
+    test("Table zone isn't expanded when modifying existing cell", () => {
+      setCellContent(model, "A4", "Something");
+      createTable(model, "A1:A3");
+      setCellContent(model, "A4", "Something Else");
+      expect(zoneToXc(model.getters.getTables(sheetId)[0].zone)).toEqual("A1:A3");
+      expect(getFilterValues(model)).toEqual([{ zone: "A1:A3", value: [] }]);
+    });
+
+    test("Table zone isn't expanded when another cell below the table had a content", () => {
+      setCellContent(model, "B4", "Something");
+      createTable(model, "A1:B3");
+      setCellContent(model, "A4", "Something Else");
+      expect(zoneToXc(model.getters.getTables(sheetId)[0].zone)).toEqual("A1:B3");
+    });
+
+    test("Table zone is expanded when another cell below the table had a a style but no content", () => {
+      setStyle(model, "B4", { fillColor: "#000000" });
+      createTable(model, "A1:B3");
+      setCellContent(model, "A4", "Something");
+      expect(zoneToXc(model.getters.getTables(sheetId)[0].zone)).toEqual("A1:B4");
+    });
+
+    test("Table zone is expanded when a cell on the row below the table but not below the table had a content", () => {
+      setCellContent(model, "B4", "Something");
+      createTable(model, "A1:A3");
+      setCellContent(model, "A4", "Something Else");
+      expect(zoneToXc(model.getters.getTables(sheetId)[0].zone)).toEqual("A1:A4");
+    });
+
+    test("Table zone isn't expanded when there was another table table below it", () => {
+      createTable(model, "A1:B3");
+      createTable(model, "B4:B6");
+      setCellContent(model, "A4", "Something");
+      expect(zoneToXc(model.getters.getTables(sheetId)[0].zone)).toEqual("A1:B3");
+    });
+
+    test("Table zone isn't expanded when there was a merge below it", () => {
+      createTable(model, "A1:B3");
+      merge(model, "B4:B6");
+      setCellContent(model, "A4", "Something");
+      expect(zoneToXc(model.getters.getTables(sheetId)[0].zone)).toEqual("A1:B3");
+    });
+  });
+
+  describe("Grid manipulation", () => {
+    let model: Model;
+    let sheetId: UID;
+    beforeEach(() => {
+      model = new Model();
+      createTable(model, "C3:F6");
+      updateFilter(model, "C3", ["C"]);
+      updateFilter(model, "D3", ["D"]);
+      updateFilter(model, "E3", ["E"]);
+      updateFilter(model, "F3", ["F"]);
```

Comment:
:angry: beforeEach with lots of setup.
I can already see myself trying to debug one of those tests because I broke something and trying to keep in my head the initial state of the spreadsheet with all those filtered values, and mentally updating my mental representation with the operations in the tests.
=> :exploding_head: :skull:

Reply:
I tend to agree with you, but in this case all this setup is actually used and tested in every test. Copying that in each test would make it easier to read each test individually, but would also means plus 100 duplicated  lines of codes